### DISCUSSION TITLE
Add support for mobile devices

### DISCUSSION THREAD

**Discussion by 0-BSCode** [Taitava/obsidian-shellcommands#442] [Ideas]
Greetings!

I understand from this [thread](https://forum.obsidian.md/t/shell-commands-plugin/23497/6) that mobile devices are currently not supported by this plugin. I verified this as well by searching for it on my phone just today. I'd like to ask if this is something on the roadmap and, if so, how far down is it? I checked the roadmap on the [documentation](https://publish.obsidian.md/shellcommands/Roadmap), but it doesn't seem to provide any info on this concern.

Just stumbled upon this plugin today trying to set up a syncing solution between my iPhone and Windows PC and I love the work! I noticed this works great with the [Commander Plugin](https://github.com/phibr0/obsidian-commander) to be able to execute custom shell commands from different parts of Obsidian's UI. I'm hoping mobile devices will be supported soon as I find myself taking notes on my phone then needing to transfer them to my PC.

**Comment by Taitava** [comment]
Hi @0-BSCode ! Thank you for your question and feedback! 🙂 

Unfortunately, mobile support is not on the roadmap. Problems:
- AFAIK, [iOS does not allow apps to execute shell commands](https://stackoverflow.com/a/72825232).
- For Andorind, I found some results on how to execute shell commands (I did just a quick search): https://www.quora.com/Can-you-run-a-shell-script-on-an-Android-phone . But this was aimed more for how a user can install and use a terminal app, not exactly about how to **develop** a terminal like app.

So, my research on this subject is very thin. If someone has better information, please enlighten me. But even if there is a way for mobile apps to do it, I don't know if it's doable by an Obsidian plugin. I'd need to learn how plugins work in Obsidian **Mobile**. For example, if a library would be needed for this, a plugin would not be able to install it, it would need to be installed by (or included in?) the main Obsidian Mobile app.

---

But even if it was possible to do, I'd probably still decide not to expand the SC plugin to that direction. Currently, SC supports three different operating systems: Windows, Linux, and macOS. It's a lot of work already for me to support them: fix bugs, help resolve problems, write documentation and examples. SC still has unique problems on all of those platforms. (And I even cannot test macOS easily at all as I don't have a Mac.)

Expanding to mobile would increase the amount of supportable operating systems to five. I'm afraid that would reserve too much my resources from developing new features and fixing current issues. And it would take a long time to get the mobile side working ok without too many bugs that always arise with huge new features.

So, unfortunately it's a big no this time. 😢

**Comment by Taitava** [comment]
Offtopic: As you mentioned that you would use SC to sync notes from iPhone to Windows, I got some ideas. I don't know how good they are for you, and maybe you already have a synchronising system in your mind that you do not wish to change, but here are some ideas anyway:
- Maybe you could use the [Git plugin](https://github.com/Vinzent03/obsidian-git) for Obsidian to do the synchronization? If I understood correctly, it should work on mobile, too. [Looks like it's using isomorphic Git](https://github.com/Vinzent03/obsidian-git/blob/master/src/gitManager/isomorphicGit.ts), so it doesn't need to execute external programs to do it's job.
- Maybe you already know this, but just in case other readers don't: [Obsidian Sync](https://obsidian.md/sync) might be the easiest solution to set up. However, it has a recurring fee, so it doesn't suit for all use-cases.
- I've seen some other community plugins providing synchronization services, too, but I can't remember names nor are they available for mobile.
